serprog.c

```diff
@@ -275,7 +275,8 @@ static int sp_stream_buffer_op(uint8_t cmd, uint32_t parmlen, uint8_t *parms)
 		return 1;
 	}
 	sp[0] = cmd;
-	memcpy(&(sp[1]), parms, parmlen);
+	if (parmlen > 0)
```

Comment:
Looking at Linux and POSIX documentation for memcpy() I can't see anything that could ever explode. If anything, this would be hiding bugs in libc.

What warning option is this about? is it part of -Wextra?

Reply:
It's a warning from clang:

    ../../../serprog.c:278:2: warning: Null pointer passed as an argument to a 'nonnull' parameter
        memcpy(&(sp[1]), parms, parmlen);

Reply:
Thanks, do you happen to know if this 'nonnull' is hardcoded in Clang or an attribute in the code? In either case, I would consider this for a bug report. Spurious warnings result in unnecessarily complex code and that in a higher human failure rate.

Reply:
Okay, agreed. I've dropped this.